### PR TITLE
Logging error handling

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/error/errorHandler.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    status: err && err.status ? err.status : 400,
+    message: err && err.message ? err.message : 'Something went wrong',
+  };
+
+  if (err && err.response) {
+    formattedError.status = err.response.status;
+    formattedError.message = `${err.response.data.errors[0]} ${err.response.statusText}`;
+    if (err.response.status === 404) {
+      const splitUrl = err.response.config.url.split('/');
+      const solutionId = splitUrl[splitUrl.length - 2];
```
Comment (vladimir32pa) [inline app/error/errorHandler.js]: should errorHandler know anything about Solution ids and its error message? If solution page url changes, that this have to be changed as well

Comment (kat1906) [reply app/error/errorHandler.js]: How generic we make the errorHandler is up for debate. I was just playing around with it. Due to the low complexity in this app I thought we could get away with this (as long as the api doesn't change it's responses). Alternatively we could ask them to return better error messages. Will discuss with everyone at later date.

File: app/error/logger.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const { createLogger, format, transports } = require('winston');
```
Comment (alibombaye) [inline app/error/logger.js]: Just a comment about the location of this `logger` file. Could it be moved somewhere else?

File: app/error/template.njk [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        "href": "/solutions/all",
+        "text": "Go to All Solutions"
+      }) }}
+    </div>
+    <div class="nhsuk-u-padding-top-5 nhsuk-grid-column-one-third" data-test-id="go-to-foundation-solutions-link"> 
+      {{ backLink({
+        "href": "/solutions/foundation",
+        "text": "Go to Foundation Solutions"
+      }) }}
+    </div>
```
Comment (alibombaye) [inline app/error/template.njk]: I don't understand why there are 3 backlinks?

Comment (kat1906) [reply app/error/template.njk]: because it is a generic error page. can be changed just did what I thought as there was no guidance

Comment (kat1906) [reply app/error/template.njk]: pull the branch and take a look

File: app/error/logger.js (hunk 2 above)
Comment (kat1906) [reply app/error/logger.js]: yes I just put it there for now so everyone could take a look and get feedback - it isn't the polished product!

Comment (alibombaye) [reply app/error/logger.js]: can we move this to the root for now. i think we are good with what you've done here.

File: app/error/template.njk (hunk 3 above)
Comment (alibombaye) [reply app/error/template.njk]: Just move these back links above the Error message

**Pull request merged**